File: update_fab.dart
import 'dart:io';

void main() {
  var dirs = [
    Directory('lib/view/screens'),
    Directory('lib/feature'),
  ];
  for (var dir in dirs) {
    if (dir.existsSync()) {
      for (var f in dir.listSync(recursive: true)) {
        if (f is File && f.path.endsWith('.dart')) {
          var content = f.readAsStringSync();
          if (content.contains('floatingActionButton: CustFab') &&
              !content.contains('floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat')) {
             var newContent = content.replaceAllMapped(
                 RegExp(r'(\s+)floatingActionButton:\s*CustFab'),
                 (match) => '${match[1]}floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,${match[1]}floatingActionButton: CustFab'
             );
             f.writeAsStringSync(newContent);
             print('Updated ${f.path}');
          }
        }
      }
    }
  }
}

File: update_fab.py
import os
import re

dirs_to_search = [
    r"c:\Users\ASUS\AndroidStudioProjects\om_appcart\lib\view\screens",
    r"c:\Users\ASUS\AndroidStudioProjects\om_appcart\lib\feature"
]

for d in dirs_to_search:
    for root, dirs, files in os.walk(d):
        for file in files:
            if file.endswith(".dart"):
                path = os.path.join(root, file)
                try:
                    with open(path, "r", encoding="utf-8") as f:
                        content = f.read()
                    
                    if "floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat" not in content and "floatingActionButton: CustFab" in content:
                        content = re.sub(
                            r"(\s+)floatingActionButton:\s*CustFab",
                            r"\1floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,\1floatingActionButton: CustFab",
                            content
                        )
                        with open(path, "w", encoding="utf-8") as f:
                            f.write(content)
                        print(f"Updated {file}")
                except Exception as e:
                    pass
